base_layer/core/src/transactions/transaction.rs

```diff
@@ -353,7 +394,38 @@ impl TransactionOutput {
 
     /// Verify that range proof is valid
     pub fn verify_range_proof(&self, prover: &RangeProofService) -> Result<bool, TransactionError> {
-        Ok(prover.verify(&self.proof.to_vec(), &self.commitment))
+        Ok(prover.verify(&self.proof.0, &self.commitment))
+    }
+
+    /// Attempt to rewind the range proof to reveal the proof message and committed value
+    pub fn rewind_range_proof(
+        &self,
+        prover: &RangeProofService,
+        public_rewind_key: &PublicKey,
```

Comment:
nit: Perhaps `rewind_public_key` and `rewind_blinding_public_key` is a better name? We don't want to suggest that these values are published somewhere